### PR TITLE
PublicDashboards: Do not return hidden queries

### LINKED ISSUE #71326: Public Dashboards: Disabled queries are still displayed in public dashboard
### 
# What went wrong?

**What happened**:

  - When I displayed the dashboard as a public dashboard, it visualized all the queries on a panel, regardless of the queries being enabled or disabled. This however is not the case when the dashboard is loaded in normal view. There, the disabled queries are indeed not displayed, as they should. 

**What did you expect to happen**:

  - When I disabled the queries, I expect them to also not be displayed on the dashboard. This worls on a regular dashboard, just not on a public dashboard.

### 
# How do we reproduce it?

1: Create a Grafana Dashboard
2: Add a data source
3: Create a panel in the Dashboard
4: Select any type of visualization (Gauge in my case)
5: Add a query to select some data from the data source
6: Add a second query to select some data from the data source
7: disable one of the two queries
- this should now only visualize one of the queries.
8: share the dashboard as a public dashboard
9: the public dashboard should now visualize both the query which was enabled, AND the query which was DISABLED.

### 
# What Grafana version are you using?

Grafana v10.0.1 (5a30620b85)

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

A downloaded binary

### User's OS?

Windows 10

### User's Browser?

Firefox

### Is this a Regression?

Yes

### Are Datasources involved?

Microsoft SQL Server

### Anything else to add?

This is how the dashboard panel looks in normal view: 2023-07-11 10_10_35-Window

This is how the queries are disabled/enabled 2023-07-11 10_11_10-Window

This is how the panel is displayed on the public dashboard: 
2023-07-11 10_10_21-Window

### PR DESCRIPTION
**Pull request by evictorero**
**What is this feature?**

Remove hidden queries from the public dashboard response

Fixes https://github.com/grafana/grafana/issues/71326

**Special notes for your reviewer:**
Hidden queries are removed from the frontend on regular dashboards but because public dashboards grab all the queries from the db we need to filter them before hitting the data sources.
There is a special case when a target contains an expression, the responsible for filter the hidden queries is the expression handler because some shown queries could need the value of a hidden query.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/publicdashboards/service/query_test.go [hunk 1]
```diff
@@ -1511,6 +1527,20 @@ func TestGroupQueriesByPanelId(t *testing.T) {
 
 		assert.Len(t, queries, 2)
 	})
+
+	t.Run("hidden queries get are not returned", func(t *testing.T) {
```
Comment (dprokop) [inline pkg/services/publicdashboards/service/query_test.go:1531]:
```suggestion
	t.Run("hidden queries are not returned", func(t *testing.T) {
```

File: pkg/services/publicdashboards/service/query.go [hunk 2]
```diff
@@ -280,11 +281,19 @@ func extractQueriesFromPanels(panels []interface{}, result map[int64][]*simplejs
 		}
 
 		var panelQueries []*simplejson.Json
+		hasExpression := panelHasAnExpression(panel)
```
Comment (dprokop) [inline pkg/services/publicdashboards/service/query.go:284]: Just a question, to understand the underlying flow of expression. Is it possible for expressions to evaluate hidden queries?

**Review by dprokop — APPROVED** [review]
Looks good

**Review by ivanortegaalba — APPROVED** [review]

File: pkg/services/publicdashboards/service/query.go (hunk 2 above)
Comment (evictorero) [reply pkg/services/publicdashboards/service/query.go:284]: Yes, an expression like `$A + $B` could be displayed even when queries `A` and `B` were hidden. This is why we can't remove them in advance before calculating the expression.

**Pull request merged**